x-pack/plugins/rollup/server/routes/api/search.js

```diff
@@ -14,14 +14,18 @@ export function registerSearchRoute(server) {
     path: '/api/rollup/search',
     method: 'POST',
     handler: async (request, reply) => {
-      const { index, query } = request.payload;
       const callWithRequest = callWithRequestFactory(server, request);
 
       try {
-        const results = await callWithRequest('rollup.search', {
-          index,
-          body: query,
-        });
+        const results = [];
+
+        for (let i = 0; i < request.payload.length; i++) {
+          const { index, query } = request.payload[i];
+          results.push(await callWithRequest('rollup.search', {
```

Comment:
using `await` in a loop blocks the loop iteration until it resolves, right? could we change this a little to use `Promise.all` instead so that the calls are truly asynchronous?

```js
        const results = [];

        for (let i = 0; i < request.payload.length; i++) {
          const { index, query } = request.payload[i];
          results.push(callWithRequest('rollup.search', {
            index,
            body: query,
          }));
        }

        reply(await Promise.all(results));
```

https://eslint.org/docs/rules/no-await-in-loop

Reply:
So funny, I went on a walk and realized the same thing, and I knew this comment was waiting for me when I came back...